### PR TITLE
Fix: don't retry connect on 404

### PR DESCRIPTION
**Pull request by sharafian**
Resolves https://github.com/interledger/js-ilp-plugin-bells/issues/24 . If an account doesn't exist on the ledger there is no point in retrying.

### CODE REVIEW

File: src/lib/plugin.js [hunk 1]
```diff
@@ -35,7 +35,10 @@ function * requestRetry (opts, errorMessage, credentials) {
         {json: true},
         requestCredentials(credentials),
         opts))
-      if (res.statusCode >= 400 && res.statusCode < 500) {
+      if (res.statusCode === 404) {
+        debug('request status 404.')
+        break // don't retry if it's a 404
+      } else if (res.statusCode >= 400 && res.statusCode < 500) {
```
Comment (emschwartz) [inline src/lib/plugin.js]: Should we be retrying on any `400` status codes? @justmoon

Comment (justmoon) [reply src/lib/plugin.js]: As a general rule, you should retry on 5xx (server messed up, maybe it'll be fixed later) and not retry on 4xx (you messed up, unless you changed something, probably no point in just trying again).

Comment (emschwartz) [reply src/lib/plugin.js]: 👍  @bensharaf can you change it to not retry on any 4xx errors?

File: src/lib/plugin.js [hunk 2]
```diff
@@ -26,31 +26,17 @@ function wait (ms) {
...
-    }
+function * requestLedger (opts, errorMessage, credentials) {
+  debug('connecting to account ' + opts.uri)
+  let res = yield request(Object.assign(
+    {json: true},
+    requestCredentials(credentials),
+    opts))
+  if (res.statusCode >= 400 && res.statusCode < 500) {
+    debug('request status ' + res.statusCode + '.')
+    throw new Error(errorMessage)
```
Comment (emschwartz) [inline src/lib/plugin.js]: Please add a test for this behavior

**Comment by justmoon** [comment]
LGTM

**Pull request merged**